<launch>
    <!-- =========================================================================================================== -->
    <!-- ===========================================NODE SETTINGS=================================================== -->
    <!-- =========================================================================================================== -->
    <!-- Global Settings For All Nodes -->
    <arg name="2D_camera_name" default="linear_3dof_2d_camera"/>
    <arg name="3D_camera_name" default="linear_3dof_3d_camera"/>
    <arg name="odom_frame_id" default="linear_3dof_arm_home"/>
    <arg name="device_sync" default="false"/>
    <arg name="initial_camera_reset" default="true"/>

    <!-- 2D Camera Settings -->
    <arg name="2D_camera_serial" default="841612070382"/>
    <arg name="2D_json_config" default="$(find grasberry_manipulation)/config/$(arg 2D_camera_name).json"/>
    <arg name="2D_rgb_max_side" default="1920"/>
    <arg name="2D_rgb_min_side" default="1080"/>
    <arg name="2D_rgb_fps" default="30"/>
    <arg name="2D_depth_max_side" default="1280"/>
    <arg name="2D_depth_min_side" default="720"/>
    <arg name="2D_depth_fps" default="30"/>
    <arg name="2D_align_depth_to_colour" default="true"/>
    <arg name="2D_imu_enabled" default="false"/>
    <arg name="2D_infrared_enabled" default="false"/>
    <arg name="2D_filters" default="disparity,spatial"/>

    <!-- 3D Camera Settings -->
    <arg name="3D_camera_serial" default="843112074204"/>
    <arg name="3D_json_config" default="$(find grasberry_manipulation)/config/$(arg 3D_camera_name).json"/>
    <arg name="3D_rgb_max_side" default="848"/>
    <arg name="3D_rgb_min_side" default="480"/>
    <arg name="3D_rgb_fps" default="6"/>
    <arg name="3D_depth_max_side" default="848"/>
    <arg name="3D_depth_min_side" default="480"/>
    <arg name="3D_depth_fps" default="6"/>
    <arg name="3D_align_depth_to_colour" default="true"/>
    <arg name="3D_imu_enabled" default="false"/>
    <arg name="3D_infrared_enabled" default="false"/>
    <arg name="3D_filters" default="disparity,spatial,temporal"/>

    <!-- =========================================================================================================== -->
    <!-- ===============================================NODES======================================================= -->
    <!-- =========================================================================================================== -->
    <!-- Launch 2D camera -->
    <include file="$(find rasberry_perception)/launch/camera_publisher.launch">
        <arg name="odom_frame_id" value="$(arg odom_frame_id)"/>
        <arg name="filters" value="$(arg 2D_filters)"/>
        <arg name="initial_reset" value="$(arg initial_camera_reset)"/>
        <arg name="camera_name" value="$(arg 2D_camera_name)"/>
        <arg name="serial_no" value="$(arg 2D_camera_serial)"/>
        <arg name="json_file_path" value="$(arg 2D_json_config)"/>
        <arg name="device_sync" value="$(arg device_sync)"/>
        <arg name="rgb_max_side" value="$(arg 2D_rgb_max_side)"/>
        <arg name="rgb_min_side" value="$(arg 2D_rgb_min_side)"/>
        <arg name="rgb_fps" value="$(arg 2D_rgb_fps)"/>
        <arg name="depth_max_side" value="$(arg 2D_depth_max_side)"/>
        <arg name="depth_min_side" value="$(arg 2D_depth_min_side)"/>
        <arg name="depth_fps" value="$(arg 2D_depth_fps)"/>
        <arg name="align_depth_to_colour" value="$(arg 2D_align_depth_to_colour)"/>
        <arg name="imu_enabled" value="$(arg 2D_imu_enabled)"/>
        <arg name="infrared_enabled" value="$(arg 2D_infrared_enabled)"/>
    </include>

    <!-- Launch 3D camera -->
    <include file="$(find rasberry_perception)/launch/camera_publisher.launch">
        <arg name="odom_frame_id" value="$(arg odom_frame_id)"/>
        <arg name="filters" value="$(arg 3D_filters)"/>
        <arg name="initial_reset" value="$(arg initial_camera_reset)"/>
        <arg name="camera_name" value="$(arg 3D_camera_name)"/>
        <arg name="serial_no" value="$(arg 3D_camera_serial)"/>
        <arg name="json_file_path" value="$(arg 3D_json_config)"/>
        <arg name="device_sync" value="$(arg device_sync)"/>
        <arg name="rgb_max_side" value="$(arg 3D_rgb_max_side)"/>
        <arg name="rgb_min_side" value="$(arg 3D_rgb_min_side)"/>
        <arg name="rgb_fps" value="$(arg 3D_rgb_fps)"/>
        <arg name="depth_max_side" value="$(arg 3D_depth_max_side)"/>
        <arg name="depth_min_side" value="$(arg 3D_depth_min_side)"/>
        <arg name="depth_fps" value="$(arg 3D_depth_fps)"/>
        <arg name="align_depth_to_colour" value="$(arg 3D_align_depth_to_colour)"/>
        <arg name="imu_enabled" value="$(arg 3D_imu_enabled)"/>
        <arg name="infrared_enabled" value="$(arg 3D_infrared_enabled)"/>
    </include>
</launch>
